zerver/views/user_settings.py

```diff
@@ -168,14 +168,18 @@ def json_set_avatar(request, user_profile):
         return json_error("You must upload exactly one avatar.")
 
     user_file = list(request.FILES.values())[0]
-    upload_avatar_image(user_file, user_profile, user_profile.email)
-    do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
-    user_avatar_url = avatar_url(user_profile)
-
-    json_result = dict(
-        avatar_url = user_avatar_url
-    )
-    return json_success(json_result)
+    try:
+        upload_avatar_image(user_file, user_profile, user_profile.email)
+        do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+
+        json_result = dict(
+            avatar_url = user_avatar_url
+        )
+        return json_success(json_result)
+    except IOError as e:
```

Comment:
What I understood is that I have to raise JsonableError in inner functions when the decode error happens, but there is a problem with that.

The inner functions are in `zerver.lib.upload`. `json_set_avatar(request, user_profile) calls
`upload_avatar_image_local(user_file, user_profile, email)` which calls `resize_avatar(image_data)`.


I didn't understand what exactly JsonableError means, but it feels slightly weird to raise JsonableError in functions which have nothing to do with json. (views which call these functions deal with json, but these functions themselves don't.)

But even with this weirdness, this idea doesn't seem bad. Should I implement it? I'll catch IOError in `resize_avatar` and raise JsonableError there.